Raven.Database/Raft/Storage/Handlers/ReplicationStateCommandHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Rachis;
+using Raven.Abstractions.Extensions;
+using Raven.Bundles.Replication.Data;
+using Raven.Database.Raft.Commands;
+using Raven.Database.Server.Tenancy;
+
+namespace Raven.Database.Raft.Storage.Handlers
+{
+    public class ReplicationStateCommandHandler : CommandHandler<ReplicationStateCommand>
+    {
+        public ReplicationStateCommandHandler(DocumentDatabase database, DatabasesLandlord landlord,RaftEngine engine) : base(database, landlord)
+        {
+            MaxReplicationLatency = database.Configuration.Cluster.MaxReplicationLatency;
+            Engine = engine;
+        }
+
+        private RaftEngine Engine { get; set; }
+
+        //default should be 2 minutes
+        public TimeSpan MaxReplicationLatency { get; set; }
+
+        public override void Handle(ReplicationStateCommand command)
+        {
+            //If we are the leader we won't have documents sent from ourself so it is a noop.
+            if (Engine.State == RaftEngineState.Leader)
```

Comment:
You _always_ need to handle it.
The state of all the nodes in the cluster after processing the commands must be the same.
You can't skip it because you are leader or not, that invalidate the whole premise of the state machine.
Leader is transient state.